papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/controller/ProjectsController.java

```diff
@@ -104,6 +110,39 @@ private Project getValidProject(Long id) {
         return ResponseEntity.ok(sprints);
     }
 
+    @Override
+    public ResponseEntity<SprintSummaryDto> getSprintSummary(Long projectId, Long sprintId) {
+        Project project = getValidProject(projectId);
+        Sprint sprint = getValidSprint(sprintId);
+
+        if (!isUserRoleAdmin(projectId, userContext.getUserIdn()) || !isSprintInProject(sprint, project)) {
+            throw new ForbiddenAccessException();
+        }
+
+        List<AvailabilityDto> availabilities = projectService
+                .getAllUsersIdInProject(project)
+                .stream()
+                .map(id -> availabilityService.getBySprintIdAndUserId(sprintId, id).map(availabilityMapper::mapToApi)
+                        .orElseGet(
+                                () -> new AvailabilityDto().userId(id).timeAvailable(0L).timeRemaining(0L).notes("")))
```

Comment:
Wiem, że to jest związane z decyzjami projektowymi w interfejsach (required i takie tam), ale miałam myśl, że w związku z tym nie odróżnimy wizualnie kogoś, kto podał 0h + 0h + "", od kogoś, kto nie podał nic (jeden jedzie na wakacje, ale nie miał na tyle honoru, żeby napisać w komentarzu, a drugi jeszcze nie podał i trzeba na niego krzyczeć). Może już była o tym jakaś dyskusja, której nie ogarnęłam czy coś, nie wiem

W sumie nie wiem, co w związku z tym. Można by zmienić strukturę klasy `AvailabilityDto` na jakieś `UserAvailability` (hejtowałam tę nazwę, wiem, ale nie mam lepszej) z polami na id oraz na `AvailabilityDto` - to drugie nullowalne

Może czegoś nie widzę, nie wiem. Jak ktoś ma jakieś przemyślenia w tej kwestii, to zapraszam do dyskusji

Reply:
Może ta flaga nie jest takim złym pomysłem, wtedy jej wartość to może być jakieś podświetlenie, a treść do wyświetlenia już jest przekazana.  To chyba najbardziej zależy od tego jak od frontu jest najwygodniej

Reply:
nope, flaga to zły pomysł. Czy ktoś, kto nie podał nic nie będzie odróżnialny przez to, że nie istnieje dla niego Availability w bazie?

Reply:
Jest odróżnialny, ale w tutaj wypluwamy dla niego `AvailabilityDto` z 0h dostępności, 0h zaległości i "" w komentarzu

Reply:
no okeeej, ale żeby taka dostępność stała się faktem, to user musi kliknąć save nie? jak nie chce uzupełniać, to kliknie anuluj i dalej będzie tym wrednym człowiekiem, który nie podał dostępności

Reply:
Nie, nie, to nie to - to jest Excel, czyli masz te wiersze z kolejnymi osobami

Reply:
I teraz trzeba będzie wyifować `(available == 0) && (remaining == 0) && (notes == "")`, żeby kogoś jakoś tak na szaro/czerwono/jakkolwiek wyświetlić, że nie podał (znaczy w sumie nie wiem, czy w Excelu osiągniemy na jutro wyświetlanie inaczej osób, które jeszcze nie podały)

Reply:
aa, to faktycznie problem. Pomysł z UserAvailability mi się osobiście najbardziej podoba i jest chyba najwygodniejszy.

Reply:
@MrMarosis Ten pomysł ma dwa głosy w takim razie - jeśli Ci pasuje, to zapraszam do tworzenia PR w interfejsach

Reply:
mamy tutaj jedno zapytanie dla kazdego usera, nie dalo by sie tego pobrac w jednym i zagregowac?
nie wiem jak to jest z bardziej zaawansowanymi zapytaniami w springu/hibernate i czy warto teraz. Ale jest to do rozwazenia na przyszlosc.

Reply:
> @MrMarosis Ten pomysł ma dwa głosy w takim razie - jeśli Ci pasuje, to zapraszam do tworzenia PR w interfejsach

https://github.com/papaya-agh/papaya-interface/pull/17